### ISSUE TITLE
sha256 not matching

### ISSUE THREAD

**Issue by vivlim** [mistydemeo/tigerbrew#561]
I installed tigerbrew on a fresh install of osx 10.4 (G4 powerbook) and all of my hashes have been incorrect.
brew doctor reports no problems and I have tried with brewed sha2 installed. I saw this bug: #316 

to get this far, I've been modifying the hashes in the recipes but this is clearly not ideal. is there a way I can change which tool calculates the hashes?

`Error: SHA256 mismatch
Expected: 83b0a317f6039ad95ef6af6a182cf659c2d053eed5d8f70b06710eb787f8aa6f
Actual: 6278f9656e6a08708588b9355f1fb05fb1d3416fa38d2d534a147ed424443c05
Archive: /Library/Caches/Homebrew/git--man-2.10.2.tar.xz
To retry an incomplete download, remove the file above.

vivtop-g4:~ vivlim$ sha2 /Library/Caches/Homebrew/git--man-2.10.2.tar.xz
SHA-256 (/Library/Caches/Homebrew/git--man-2.10.2.tar.xz) = 83b0a317f6039ad95ef6af6a182cf659c2d053eed5d8f70b06710eb787f8aa6f
SHA-384 (/Library/Caches/Homebrew/git--man-2.10.2.tar.xz) = 5c50e477643403eeb0ced8a8a7816927424814d327568ceb8ff55f5ad4c6d53661b9419b7f5232e0143413c7e5062b64
SHA-512 (/Library/Caches/Homebrew/git--man-2.10.2.tar.xz) = 0dc5df8441601865511983b85762102396cbf7c704fd831eb6666ef7b7639252b971f6b4080b8a57226a86124edcdd77acf24374da22047bd315769a09109bef
vivtop-g4:~ vivlim$
`

**Comment by vivlim** [comment]
found this commit where hashing was changed to the ruby internal one -- https://github.com/mistydemeo/tigerbrew/commit/1964065a71ff76a91def25978ff9146236ff7d74

I tried to write a ruby script which would calculate the sha256, but I don't know ruby so I didn't get anywhere with that.

I seem to have the right version of ruby:
vivtop-g4:~ vivlim$ ruby --version
ruby 1.8.2 (2004-12-25) [powerpc-darwin8.0]
vivtop-g4:~ vivlim$ which ruby
/usr/bin/ruby
vivtop-g4:~ vivlim$

**Comment by mistydemeo** [comment]
This got resolved, right? Mind if I close it?

**Comment by buttkus** [comment]
This is still an issue for me, same thing fresh install on 10.4 with xtools 2.5 update from apples site. Every single SHA for every package is declared a mismatch.

**Comment by buttkus** [comment]
Let me know if I can provide anything that could help.

**Comment by buttkus** [comment]
From a fresh 10.4, I installed 10.4.11, xtools 2.5, and installed tigerbrew. I had to manually edit the SHAs in the various formulas until i was able to install git and curl installed. After that I ran brew update, rebooted, and tried another formula and still didn't calculate correctly. Thinking it could be the PPC I downloaded the file also on another computer and manually moved it to the cache. Also didn't work. I know this worked on 10.4 before did something change in how the SHAs are being calculated?

**Comment by vivlim** [comment]
This resolved for me on its own, I have no idea why. I left my 10.4 mac alone for a couple days and when I picked it up again I was able to brew update & install packages successfully.

**Comment by EricFromCanada** [comment]
Just started getting SHA256 mismatches. For any Sourceforge-hosted archives, the actual file downloaded was this:
```
<html><head>
<title>SourceForge</title>
<!-- <script src="/js/jquery.com/jquery-1.11.0.min.js"></script> -->
<script src="https://code.jquery.com/jquery-1.11.0.min.js"></script>
<script src="https://sourceforge.net/js/mirrors.js"></script>
<script src="/js/sf.js"></script>
<script>
var DR_loc = DR_parse_hash_url();
if (DR_loc) {
    DR_sf_main(DR_loc);
} else {
    window.location.href = 'https://sourceforge.net/home.html';
}
</script>
</head><body>
<noscript>
We're sorry -- the Sourceforge site is currently in Disaster Recovery mode, and currently requires
the use of javascript to function.  Please check back later.
</noscript>
</body></html>
```

So, at this moment I get the wrong file if I run `curl -OL https://ssl.icu-project.org/files/icu4c/55.1/icu4c-55_1-src.tgz`, but punching that URL into a browser redirects and downloads properly.

**Comment by mistydemeo** [comment]
@EricFromCanada I get the same results when trying to download the file you linked from my own machine - it's not Tigerbrew-specific. I'll see if I can repro it with anything else... And I'll see if I can fix any broken URLs.

**Comment by mistydemeo** [comment]
With help from @luigi30, I tracked this down to the system Ruby's sha256 implementation being broken on Tiger on G4 (and possibly G3) processors. It works on G4e, G5 and Intel, which is my range of hardware.

I'm going to be working on a fix for this - sorry for the inconvenience.

**Comment by mistydemeo** [comment]
This should be fixed as of #587. Please let me know if you have trouble pulling in those changes to fix the issue!
